### ISSUE TITLE
Error: That item is unavailable. 

### ISSUE THREAD

**Issue by dimitris4** [dooboolab/react-native-iap#2317]
> Please use the [Discussion board](https://github.com/dooboolab/react-native-iap/discussions) if you want to get some help. Please use issues to report bugs.

**Description**

I am trying to upgrade from react-native-iap v.7 to 10.1.3.
IAP.getSubscriptions({ skus: subscriptionIds }) returns the subscriptions.

But when I call:
IAP.requestSubscription({
    sku: skuObj.sku,
    ...(skuObj.offerToken && {
      subscriptionOffers: [{ sku: skuObj.sku, offerToken: skuObj.offerToken }],
    }),
  })

I am getting the following Google Play error: 
**_react native iap message:That item is unavailable.,debugMessage:,code:E_ITEM_UNAVAILABLE,responseCode:4_**

**Expected Behavior**

I expect the purchase to complete successfully.

**Screenshots**

**Environment:**

- react-native-iap: 10.1.3
- react-native: 0.71.1
- Platforms (iOS, Android, emulator, simulator, device): iOS works fine, this issue occurs on android real device.

---

[Optional] **Additional Context**

**Comment by andresesfm** [comment]
Why not update to the latest?

**Comment by Prathameshlabde** [comment]
https://github.com/dooboolab-community/react-native-iap/issues/2455 Can anyone please suggest on this issue Please ?

**Comment by Kishore-D18** [comment]
Have you got any solutions @Prathameshlabde

**Comment by Matiassimone** [comment]
Any update ? 
In my case is sporadically error, sometime appears and other one not.

Maybe its related to this ticket ?
https://github.com/dooboolab-community/react-native-iap/issues/1593

**Comment by Matiassimone** [comment]
I can replicate the error with consumable items, in iOS & Android.

**Comment by Dipak-Complitech** [comment]
Yes i received error like   Purchase error: {"code": "E_ITEM_UNAVAILABLE", "debugMessage": "", "message": "That item is unavailable.", "responseCode": 4}

any solution for these error ?

**Comment by bjarke-uptime** [comment]
Anyone ever managed to fix this one? I'm currently testing on an Android device and I have ensured that my Google account is added in the "License testing"-part of the Developer platform. I'm just getting 

`{"code": "E_ITEM_UNAVAILABLE", "debugMessage": "", "message": "That item is unavailable.", "responseCode": 4}`

It's working just fine on iOS.

**Comment by Dipak-Complitech** [comment]
Yes it is working fine in ios and now it's working in android also

**Comment by bjarke-uptime** [comment]
> Yes it is working fine in ios and now it's working in android also

Did you need to do any additional setup of your Google Developer account to get it working? Or perhaps there's some setup of the library I'm missing?

**Comment by Dipak-Complitech** [comment]
> > Yes it is working fine in ios and now it's working in android also
> 
> Did you need to do any additional setup of your Google Developer account to get it working? Or perhaps there's some setup of the library I'm missing?

New Lib having some method changes like

const products = await getSubscriptions({skus:itemSkus});
const offerToken = product.subscriptionOfferDetails[0].offerToken

 await requestSubscription({
          sku: this.state.selectSubscription,
          ...(offerToken && {
             subscriptionOffers: [
               {
                 sku: this.state.selectSubscription,
                 offerToken: offerToken,
               },
             ],
          }),
        })
        .catch(err => {
           console.log('error buying product', err);
         })

**Comment by hyochan** [comment]
I’m closing all issues reported in versions below 14, as the library now supports the new architecture with NitroModules and has been completely revamped.
